src/mpyl/steps/deploy/k8s/chart.py

```diff
@@ -350,6 +354,22 @@ def to_prometheus_rule(self, alerts: list[Alert]) -> V1PrometheusRule:
             alerts=alerts,
         )
 
+    def to_service_monitor(self) -> V1ServiceMonitor:
+        metrics = self.deployment.kubernetes.metrics
+        endpoint = {
+            "honorLabels": "true",
+            "path": metrics.path or "/metrics",
+            "targetPort": metrics.port or self.__find_default_port(),
+        }
```

Comment:
I think it I would find it cleaner to keep the `V1ServiceMonitor` object construction logic inside the constructor. And pass in what's necessary. So `metrics` and the default port.